pkg/util/strategicpatch/patch.go

```diff
@@ -31,7 +31,352 @@ import (
 // lists should be merged or replaced.
 //
 // For more information, see the PATCH section of docs/api-conventions.md.
+
+// CreateStrategicMergePatch creates a patch that can be passed to StrategicMergePatch.
```

Comment:
Agreed that we need a mode that detects changes and deletions, while ignoring additions. I'm thinking that it's a method called ComputeChangesAndDeletions that returns the set of changes and deletions. The check for divergence can be made by calling this method, and then looking at the result. If and only if it's empty, then there were no changes or deletions between original and modified.